Fix build failure with glibc 2.26.

Patch copied from upstream source repository:
https://sourceware.org/git/?p=valgrind.git;a=commit;h=2b5eab6a8db1b0487a3ad7fc4e7eeda6d3513626

diff --git a/memcheck/tests/linux/stack_changes.c b/memcheck/tests/linux/stack_changes.c
index a978fc2..7f97b90 100644
--- a/memcheck/tests/linux/stack_changes.c
+++ b/memcheck/tests/linux/stack_changes.c
@@ -10,7 +10,7 @@
 // This test is checking the libc context calls (setcontext, etc.) and
 // checks that Valgrind notices their stack changes properly.
 
-typedef  struct ucontext  mycontext;
+typedef  ucontext_t  mycontext;
 
 mycontext ctx1, ctx2, oldc;
 int count;
